opal/util/info_subscriber.c

```diff
@@ -266,12 +255,11 @@ int opal_infosubscribe_change_info(opal_infosubscriber_t *object, opal_info_t *n
             key_str = iterator->ie_key;
             OBJ_RETAIN(key_str);
 
-            updated_value = opal_infosubscribe_inform_subscribers(object, iterator->ie_key->string,
-                                                                  iterator->ie_value->string,
-                                                                  &found_callback);
-            if (NULL != updated_value
-                && 0 != strncmp(updated_value, value_str->string, value_str->length)) {
-                err = opal_info_set(object->s_info, iterator->ie_key->string, updated_value);
+            update_cnt = opal_infosubscribe_inform_subscribers(object, iterator->ie_key->string,
+                                                               iterator->ie_value->string);
+            if (0 < update_cnt) {
+                err = opal_info_set(object->s_info, iterator->ie_key->string,
+                                    iterator->ie_value->string);
```

Comment:
>  I understand what the problem is, but I don't understand the proposed solution. Why is there a need to duplicate the info before updating them ?

@bosilca To me the confusion is actually here - is `opal_infosubscribe_change_info` supposed to also update `object->s_info` or not. Correct me if I'm wrong, but the code reads to me that `opal_infosubscribe_change_info` is only responsible for notifying the 'subscribers' of `new_info`. L260-L262 somewhat looks out of place.

This leads me to think that setting `object->s_info`(via `opal_info_dup(info, &newcomp->super.s_info)`) should be separate from `opal_infosubscribe_change_info`.

Please let me know what I missed.

Reply:
For what the extra information might be worth, there were `opal_info_dup` calls present before 6bd36a7:

https://github.com/open-mpi/ompi/blob/3b4d64ffef2bd30a381bdd8e9e909be821955b1b/ompi/communicator/comm.c#L613-L617

Reply:
After digging into the code it seems the root cause was a lack of validation and a lack of understanding of the impact of 6bd36a7 into the code base. This issue with the info is just one of the many other issues that got into the main, in exchange of being able to create communicators with a tag given by an external software stack (aka to create sessions).

As a result, I retract my original complain about duplicating the info. I still have an issue with the changes in `opal/util/info_subscriber.c`.